### PR TITLE
Add CNN torch-native StarPU ops (conv, pool, batch_norm)

### PR DESCRIPTION
**Pull request by Muxas**
## Summary

Extends `device=nntile` StarPU PrivateUse1 coverage for CNN training paths, plus tiny sample scripts and a measured CPU vs nntile showcase.

### CNN aten / StarPU stack
- Pool / conv / BN PrivateUse1 ops (AvgPool, AdaptiveAvgPool, MaxPool+indices, Convolution, NativeBatchNorm) + practical packing / contiguous-grad fixes.
- **New:** `upsample_nearest2d` / `upsample_bilinear2d` (+ backward) via StarPU `TorchKind::Upsample*` unary path (`nntile_upsample2d.cpp`).

### Sample training scripts
- Classification: LeNet, ResNet, VGG, MobileNet
- Segmentation: classic U-Net (`ConvTranspose2d`) and **modern U-Net** (`F.interpolate` bilinear/nearest)
- Showcase: [`docs/dev/cnn_tiny_cpu_vs_nntile_showcase.md`](docs/dev/cnn_tiny_cpu_vs_nntile_showcase.md)

### Measured results (`ncpu=1`, `steps=1`, `batch-size=2`, `seed=0`)

| Model | CPU loss | nntile loss | CPU wall (s) | nntile wall (s) | Δ loss | Status |
|---|---:|---:|---:|---:|---:|---|
| lenet | 2.230573 | 2.230573 | 0.005 | 0.011 | 0.000e+00 | OK |
| resnet | 1.905449 | 1.905449 | 0.005 | 0.016 | 0.000e+00 | OK |
| vgg | 2.474704 | 2.474704 | 0.007 | 0.013 | 0.000e+00 | OK |
| mobilenet | 2.080944 | 2.080944 | 0.006 | 0.017 | 0.000e+00 | OK |
| unet | 1.160912 | 1.160912 | 0.016 | 0.070 | 0.000e+00 | OK |
| unet_modern | 1.164545 | 1.164545 | 0.015 | 0.043 | 0.000e+00 | OK |

## Test plan
- [x] `bench_cnn_tiny_cpu_vs_nntile.py` — all six models, losses match
- [x] Modern U-Net nntile train via `F.interpolate`
- [ ] CI / `aten_ops_parity` upsample cases

<div><a href="https://cursor.com/agents/bc-019f74f3-fe34-74c1-a069-c4ab2b71a3ce"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-019f74f3-fe34-74c1-a069-c4ab2b71a3ce"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

**Comment by Muxas** [comment]
@cursor review
